main/display/wsgCanvas.c

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file wsgCanvas.c
+ * @author Jeremy Stintzcum (jeremy.stintzcum@gmail.com)
+ * @brief Provides a canvas to paint with low memory requirements
+ * @version 1.0
+ * @date 2025-09-02
+ *
+ * @copyright Copyright (c) 2025
+ *
+ */
+
+//==============================================================================
+// Includes
+//==============================================================================
+
+#include <string.h> // For memcpy()
+
+#include <esp_heap_caps.h>
+
+#include "fs_wsg.h"
+#include "wsgCanvas.h"
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+void canvasBlankInit(wsg_t* canvas, int width, int height, paletteColor_t startColor, bool spiRam)
+{
+    canvas->h = height;
+    canvas->w = width;
+    // FIXME: idf.py build can't find the "tag" variant. Not sure if that's important or not
+    /* canvas->px = (paletteColor_t*)heap_caps_malloc_tag(sizeof(paletteColor_t) * canvas->w * canvas->h,
+                                                       spiRam ? MALLOC_CAP_SPIRAM : MALLOC_CAP_8BIT, "wsg"); */
+    canvas->px = (paletteColor_t*)heap_caps_malloc(sizeof(paletteColor_t) * canvas->w * canvas->h,
+                                                   spiRam ? MALLOC_CAP_SPIRAM : MALLOC_CAP_8BIT);
+    for (int idx = 0; idx < (height * width); idx++)
+    {
+        canvas->px[idx] = startColor;
+    }
```

Comment:
This is gonna be both faster and more concise:
```suggestion
    memset(canvas->px, startColor, sizeof(paletteColor_t) * canvas->w * canvas->h);
```